pkg/operator/staleconditions/remove_stale_conditions.go

```diff
@@ -2,50 +2,70 @@ package staleconditions
 
 import (
 	"context"
+	"fmt"
 	"time"
 
-	operatorv1 "github.com/openshift/api/operator/v1"
-
+	"github.com/openshift/library-go/pkg/apiserver/jsonpatch"
 	"github.com/openshift/library-go/pkg/controller/factory"
 	"github.com/openshift/library-go/pkg/operator/events"
 	"github.com/openshift/library-go/pkg/operator/v1helpers"
 )
 
 type RemoveStaleConditionsController struct {
-	conditions     []string
-	operatorClient v1helpers.OperatorClient
+	controllerInstanceName string
+	conditions             []string
+	operatorClient         v1helpers.OperatorClient
 }
 
 func NewRemoveStaleConditionsController(
+	instanceName string,
 	conditions []string,
 	operatorClient v1helpers.OperatorClient,
 	eventRecorder events.Recorder,
 ) factory.Controller {
 	c := &RemoveStaleConditionsController{
-		conditions:     conditions,
-		operatorClient: operatorClient,
+		controllerInstanceName: factory.ControllerInstanceName(instanceName, "RemoveStaleConditions"),
+		conditions:             conditions,
+		operatorClient:         operatorClient,
 	}
 	return factory.New().
 		ResyncEvery(time.Minute).
 		WithSync(c.sync).
 		WithInformers(operatorClient.Informer()).
 		ToController(
-			"RemoveStaleConditionsController", // don't change what is passed here unless you also remove the old FooDegraded condition
+			c.controllerInstanceName,
 			eventRecorder.WithComponentSuffix("remove-stale-conditions"),
 		)
 }
 
 func (c RemoveStaleConditionsController) sync(ctx context.Context, syncContext factory.SyncContext) error {
```

Comment:
would you mind adding some unit tests for the sync function ?

Reply:
Added some unit tests (bases on the node status controller)